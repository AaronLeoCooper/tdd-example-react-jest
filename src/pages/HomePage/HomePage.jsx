import React from 'react';
import PropTypes from 'prop-types';

import UsersListContainer from '../../containers/UsersListContainer';

function HomePage({ location }) {
  return (
    <div>
      <header>
        You're at: {location.pathname}
      </header>
      <UsersListContainer />
    </div>
  );
}

HomePage.propTypes = {
  location: PropTypes.shape({
    pathname: PropTypes.string.isRequired
  }).isRequired
};

export default HomePage;
