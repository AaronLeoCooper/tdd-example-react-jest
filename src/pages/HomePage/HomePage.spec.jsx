import React from 'react';
import { shallow } from 'enzyme';

import HomePage from './HomePage';

describe('HomePage', () => {
  it('Should match rendered snapshot', () => {
    const wrapper = shallow(
      <HomePage location={{ pathname: '/test' }}/>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
