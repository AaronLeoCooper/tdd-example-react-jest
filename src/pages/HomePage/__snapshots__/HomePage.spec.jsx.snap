// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage Should match rendered snapshot 1`] = `
<div>
  <header>
    You're at: 
    /test
  </header>
  <Connect(UsersListContainer) />
</div>
`;
